### PR TITLE
[Feature] Old logs collector

### PR DESCRIPTION
**Pull request by abelgardep**
## Related Issues
App: #3328 

- [x] Added changelog files for the fixed issues in folder changelog/unreleased. More info [here](https://github.com/owncloud/android/tree/master/changelog#create-changelog-items)
_____

### CODE REVIEW

File: owncloudApp/src/main/java/com/owncloud/android/workers/OldLogsCollectorWorker.kt [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+) {
+
+    override suspend fun doWork(): Result {
+        val logsDirectory = getLogsDirectory()
+        val logsFiles = getLogsFiles(logsDirectory)
+
+        return try {
+            removeOldLogs(logsFiles)
+            Result.success()
+        } catch (ioException: IOException) {
```
Comment (theScrabi) [inline owncloudApp/src/main/java/com/owncloud/android/workers/OldLogsCollectorWorker.kt:47]: If IOException and SecurityException are the only two exceptions you could combine both cases to a simple
``` kotlin
catch(exception: Exception) {
   Result.failure()
}
```

File: owncloudApp/src/main/java/com/owncloud/android/workers/OldLogsCollectorWorker.kt [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+    }
+
+    private fun getLogsDirectory(): File {
+        val localStorageProvider = LocalStorageProvider.LegacyStorageProvider(MainApp.dataFolder)
+        val logsPath = localStorageProvider.getLogsPath()
+        return File(logsPath)
+    }
+
+    private fun getLogsFiles(logsFolder: File): List<File> {
+        return logsFolder.listFiles()?.toList() ?: listOf()
```
Comment (theScrabi) [inline owncloudApp/src/main/java/com/owncloud/android/workers/OldLogsCollectorWorker.kt:61]: Also you might want to catch the NPE exception which could be triggered here.

**Review by theScrabi — APPROVED** [review]

File: owncloudApp/src/main/java/com/owncloud/android/workers/OldLogsCollectorWorker.kt (hunk 2 above)
Comment (abelgardep) [reply owncloudApp/src/main/java/com/owncloud/android/workers/OldLogsCollectorWorker.kt:61]: No NPE can be triggered here. Kotlin syntax 👀   This would be the analog code, maybe it is easier to understand:

```
if (logsFolder.listFiles() != null) {
      return logsFolder.listFiles().toList()
} else {
      return listOf()
}
```

File: owncloudApp/src/main/java/com/owncloud/android/workers/OldLogsCollectorWorker.kt (hunk 1 above)
Comment (abelgardep) [reply owncloudApp/src/main/java/com/owncloud/android/workers/OldLogsCollectorWorker.kt:47]: I agree in part. 

Catching them separately lets us handle each error in a different way. And we know exactly which are the potential errors.

In this case, that we don't do anything specific for each error, we could unify them, but I would keep them separately anyway.

**Comment by jesmrec** [comment]
In order to test this, i will cheat the following value: 

https://github.com/owncloud/android/blob/0276e4e1e26350d9fbd2ed5eb0c9bad8ecf83ef1/owncloudApp/src/main/java/com/owncloud/android/workers/OldLogsCollectorWorker.kt#L82

setting a lower frequency for the worker

Just checking that after the correct time the correct files are deleted should be enough. No corner cases.

**Comment by jesmrec** [comment]
These were the specific tests:

cheated code:

```
    companion object {
        const val OLD_LOGS_COLLECTOR_WORKER = "OLD_LOGS_COLLECTOR_WORKER"
        const val repeatInterval: Long = 2L
        val repeatIntervalTimeUnit: TimeUnit = TimeUnit.HOURS
        private const val numberOfDaysToKeepLogs = 2
    }
```

**Test 1:** 

1. Enable `Logging` in the app
2. In Device settings' , set manually a date 5 days earlier than today
3. Open the app and do something to generate logs 
4. Repeat 2. and 3., setting dates 4, 3, 2, 1 day earlier. In this way, we generate 5 files with different dates.
5. Set correct date in device settings'

After 2h, all log files were removed but the last two (newer than 2 days). ✅ 

**Test 2:** 

1. Enable `Logging` in the app
2. In Device settings' , set manually a date 5 days earlier than today
3. Open the app and do something to generate logs 
4. Repeat 2. and 3., setting dates 4, 3, 2, 1 day earlier. In this way, we generate 5 files with different dates.
5. In Device settings', set a date 4 days in the future.

After 2h, all log files were removed  ✅ 


**Test 3:** 

1. Enable `Logging` in the app
2. In Device settings' , set manually a date 5 days earlier than today
3. Open the app and do something to generate logs 
4. Repeat 2. and 3., setting dates 4, 3, 2, 1 day earlier. In this way, we generate 5 files with different dates.
5. In Device settings', set a date 3 days in the past.

After 2h, no files were removed ✅

**Comment by jesmrec** [comment]
this is approved on my side.
